Add scripts to automate the AML and TAA registration and update process

WadeBarnes opened the pull request:
Signed-off-by: Wade Barnes <wade@neoterictech.ca>

ianco commented on manage:
```diff
@@ -101,6 +101,43 @@ usage () {
+              taaRatificationTime
+                - The date and time of TAA ratification by network government.
+
+        Example:
```
Should this example work?  When I try (against a local von-network) I get:

```
$ ./manage apply-taa                 walletName=local_net_trustee_wallet                 poolName=local_net                 useDid=V4SGRU86Z58d6TV7PBUe6f                 amlUrl='https://raw.githubusercontent.com/wiki/bcgov/bc-vcpedia/(Layer-1)-CANdy-Acceptance-Mechanism-List-(AML).md'                 amlVersion=0.1                 taaUrl='https://raw.githubusercontent.com/wiki/bcgov/bc-vcpedia/(Layer-1)-CANdy-Transaction-Author-Agreement.md'                 taaVersion=0.1                 taaRatificationTime="2022-08-09T11:05:00-0700"

Using: docker-compose --log-level ERROR

date: illegal option -- -
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ... 
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
sed: 1: "/{/,/}/{/{/!{/}/!p}}": extra characters at the end of p command
Creating von_client_run ... done
Can't read the file
Goodbye...
```

WadeBarnes replied:
Yes, it should work.  That is the command I was using for testing.

WadeBarnes replied:
Do you get the same error if you run this command:
```
date --date="2022-08-09T11:05:00-0700" +"%s"
```

ianco replied:
> Do you get the same error if you run this command?
> 
> ```
> date --date="2022-08-09T11:05:00-0700" +"%s"
> ```
> 
> The expected result is `1660068300`

Yes same error, I'll try your mac version

WadeBarnes replied:
Should be fixed in the most recent commit.

swcurran commented:
With that command, I get an error on a Mac, works on Linux.

WadeBarnes commented:
The syntax on Mac is different.  it's will be something like:
```
date -jf "%Y-%m-%dT%H:%M:%S%Z" "2022-08-09T11:05:00-0700" +%s
```

Would one of you be able to confirm?

ianco commented:
> date -jf "%Y-%m-%dT%H:%M:%S%Z" "2022-08-09T11:05:00-0700" +%s

```

ianco commented:
This works, I think it automatically pucks up the local timezone:

`date -j -f "%Y-%m-%dT%H:%M:%S" "2022-08-09T11:05:00" +%s`

WadeBarnes commented:
It would be nice to be able to specify the actual time offset since the ratification time and the person running the script may not be in the same time zone.

WadeBarnes commented:
Looks like `Z%` for the MAC command needs to be an alphabetic time zone abbreviation.

So something like this should work:
`date -jf "%Y-%m-%dT%H:%M:%S %Z" "2022-08-09T11:05:00 PDT" +%s`

Expected result is `1660068300`

@ianco, would you be able to try this out when you have a moment?

swcurran commented:
Works for Mac, not for Linux/gitBash (as expected...).

WadeBarnes commented:
Turns out Linux (and Git Bash) can use the "%Y-%m-%dT%H:%M:%S %Z" format without complaint too.  So I've updated the script to support Mac and updated the example and docs to use the the "%Y-%m-%dT%H:%M:%S %Z" date format.

ianco commented on manage:
```diff
@@ -516,6 +562,65 @@ function generateGenesisFiles() {
+  curl -s -o ${taaPath} "${taaUrl}"
+
+  # Parse the AML content from the file;the bit inside the '{}'s
+  amlContent="{$(sed -n '/{/,/}/{/{/!{/}/!p}}' ${amlPath} | sed 's/^ *//g')}"
```
This is the Mac variant:  `sed -n '/{/,/}/{/{/!{/}/!p;};}' ${amlPath}`

WadeBarnes replied:
That works in git bash too.  Code updated.

WadeBarnes commented on manage:
```diff
@@ -101,6 +101,52 @@ usage () {
+                    - "2022-08-09T11:05:00 PDT"
+                - On Mac, the date format is "%Y-%m-%dT%H:%M:%S %Z", where `%Z` is an alphabetic time zone abbreviation.
+                  - Example:
+                    - "2022-08-09T11:05:00 PDT"
```
Code updated to reflect this.

WadeBarnes commented:
Code updated based on @ianco's review.

ianco reviewed (approved):
Added local file ./tmp/cliconfig.json

./manage cli init-pool local_net http://host.docker.internal:9000/genesis

./manage indy-cli create-wallet walletName=local_net_trustee_wallet

./manage apply-taa                 walletName=local_net_trustee_wallet                 poolName=local_net                 useDid=V4SGRU86Z58d6TV7PBUe6f                 amlUrl='https://raw.githubusercontent.com/wiki/bcgov/bc-vcpedia/(Layer-1)-CANdy-Acceptance-Mechanism-List-(AML).md'                 amlVersion=0.1                 taaUrl='https://raw.githubusercontent.com/wiki/bcgov/bc-vcpedia/(Layer-1)-CANdy-Transaction-Author-Agreement.md'                 taaVersion=0.1                 taaRatificationTime="2022-08-09T11:05:00PDT"
